### PR TITLE
Feature/simulate

### PR DESCRIPTION
**Pull request by jstuczyn**
This PR introduces option to simulate transaction execution and as a result ability to set an "auto" gas estimation.

Addresses https://github.com/nymtech/nym/issues/2076

### CODE REVIEW

File: common/client-libs/validator-client/src/nymd/cosmwasm_client/types.rs [hunk 1]
```diff
@@ -215,6 +219,117 @@ impl TryFrom<ProtoContractCodeHistoryEntry> for ContractCodeHistoryEntry {
...
+    pub events: Vec<abci::Event>,
+}
+
+impl TryFrom<ProtoAbciResult> for AbciResult {
+    type Error = NymdError;
+
+    fn try_from(value: ProtoAbciResult) -> Result<Self, Self::Error> {
+        // annoyingly parsing of the event could not be moved to a separate function as protobuf
+        // representation of the event is not exposed by the existing libraries
+        Ok(AbciResult {
```
Comment (durch) [inline common/client-libs/validator-client/src/nymd/cosmwasm_client/types.rs:292]: I'd maybe extract the `events` construction into its own binding, make it a bit more readable.

**Review by durch — APPROVED** [review]
Looks good in general, I'd just write out that events mapping and remapping.

File: common/client-libs/validator-client/src/nymd/cosmwasm_client/types.rs (hunk 1 above)
Comment (jstuczyn) [reply common/client-libs/validator-client/src/nymd/cosmwasm_client/types.rs:292]: I really wanted to do it, unfortunately that would require pulling separate tendermint-proto dependency just for that single type definition (it's not publicly re-exposed by cosmrs)

**Comment by durch** [comment]
`nym-wallet` also needs the `connect_with_signer()` call updated

**Review by neacsu — APPROVED** [review]

**Pull request merged**